server/types/oid/iterate.go

```diff
@@ -0,0 +1,615 @@
+	// Function is the callback for functions.
+	Function func(ctx *sql.Context, function IterationItemFunction) (cont bool, err error)
+	// Index is the callback for indexes.
+	Index func(ctx *sql.Context, schema IterationItemSchema, table IterationItemTable, index IterationItemIndex) (cont bool, err error)
+	// Schema is the callback for schemas/namespaces.
+	Schema func(ctx *sql.Context, schema IterationItemSchema) (cont bool, err error)
+	// Sequence is the callback for sequences.
+	Sequence func(ctx *sql.Context, schema IterationItemSchema, sequence IterationItemSequence) (cont bool, err error)
+	// Table is the callback for tables.
+	Table func(ctx *sql.Context, schema IterationItemSchema, table IterationItemTable) (cont bool, err error)
+	// View is the callback for views.
+	View func(ctx *sql.Context, schema IterationItemSchema, view IterationItemView) (cont bool, err error)
+	// SearchSchemas represents the search path. If left empty, then all schemas are iterated over. If supplied, then
+	// schemas are iterated by their given order.
+	SearchSchemas []string
+}
+
+// IterationItemCheck contains the relevant information to pass to the Check callback.
+type IterationItemCheck struct {
+	Index int
+	OID   uint32
+	Item  sql.CheckDefinition
+}
+
+// IterationItemForeignKey contains the relevant information to pass to the ForeignKey callback.
+type IterationItemForeignKey struct {
+	Index int
+	OID   uint32
+	Item  sql.ForeignKeyConstraint
+}
+
+// IterationItemFunction contains the relevant information to pass to the Function callback.
+type IterationItemFunction struct {
+	Index int
+	OID   uint32
+	Item  sql.Function
+}
+
+// IterationItemIndex contains the relevant information to pass to the Index callback.
+type IterationItemIndex struct {
+	Index int
+	OID   uint32
+	Item  sql.Index
+}
+
+// IterationItemSchema contains the relevant information to pass to the Schema callback.
+type IterationItemSchema struct {
+	Index int
+	OID   uint32
+	Item  sql.DatabaseSchema
+}
+
+// IterationItemSequence contains the relevant information to pass to the Sequence callback.
+type IterationItemSequence struct {
+	Index int
+	OID   uint32
+	Item  *sequences.Sequence
+}
+
+// IterationItemTable contains the relevant information to pass to the Table callback.
+type IterationItemTable struct {
+	Index int
+	OID   uint32
+	Item  sql.Table
+}
+
+// IterationItemView contains the relevant information to pass to the View callback.
+type IterationItemView struct {
+	Index int
+	OID   uint32
+	Item  sql.ViewDefinition
+}
+
+// IterateCurrentDatabase iterates over the current database, calling each callback as the relevant items are iterated
+// over. This is a central function that homogenizes all iteration, since OIDs depend on a deterministic iteration over
+// items. This function should be expanded as we add more items to iterate over.
+func IterateCurrentDatabase(ctx *sql.Context, callbacks IterationCallbacks) error {
```

Comment:
Dawg you're killing me, this is >200 lines with 8 levels of indentation, extract some methods